src/main/java/sirius/kernel/commons/Hasher.java

```diff
@@ -157,6 +162,40 @@ public Hasher hashBytes(byte[] data, int offset, int length) {
         return this;
     }
 
+    /**
+     * Reads the given inputStream and appends its bytes to the data to be hashed.
+     *
+     * @param inputStream the stream to be read and put into the hash function
+     * @return the hasher itself for fluent method calls
+     * @throws IOException in case of an IO error while reading the stream
+     */
+    public Hasher hashStream(InputStream inputStream) throws IOException {
+        checkState();
+        byte[] buffer = new byte[8192];
+
+        int readBytes;
+        do {
+            readBytes = inputStream.read(buffer);
+            if (readBytes > 0) {
+                this.digest.update(buffer, 0, readBytes);
+            }
+        } while (readBytes != -1);
+        inputStream.close();
```

Comment:
Shouldn‘t we let the caller close the stream? Zo encourage using try-with-resources